bitnami/elasticsearch/templates/master-statefulset.yaml

```diff
@@ -109,6 +109,20 @@ spec:
               value: "yes"
             - name: ELASTICSEARCH_NODE_TYPE
               value: "master"
+            {{- if .Values.extraEnvVars }}
+            {{- include "elasticsearch.tplValue" ( dict "value" .Values.extraEnvVars "context" $ ) | nindent 12 }}
+            {{- end }}
+          {{- if or .Values.extraEnvVarsConfigMap .Values.extraEnvVarsSecret }}
```

Comment:
Does it make sense for all the nodes to share the same env vars? Sometimes we've seen issues with that. In other charts we divide it by component type. However, if it makes sense maybe we can have a `commonExtraEnvVars` or, if for example all nodes require a plugin, maybe set the Plugins env var in the template.

Reply:
I agree that it would be the ideal situation, but for most cases this will be enough. I want to avoid making this a major version change of the chart, so I think this should be left out-of-scope.

The problem would come when someone wanted to change an Elasticsearch clustering flag, which is already performed with the chart. In that cases what you say would be required, but that means we should do it for everything: init scripts, environment variables **and volumes**. Currently volumes are mounted equal for all Elasticsearch containers, and changing that would require us to increase the major version.